docs/custom_models/umat_elasticity.md

```diff
@@ -0,0 +1,131 @@
+# Linear elasticity as an Abaqus UMAT
+
+All files for this example can be found in the [GitHub repository](https://github.com/BAMresearch/fenics-constitutive/tree/main/examples/umat).
+
+## Setting up the project
+
+In the preceeding examples, we have implmented a linear elasticity model using one language and looped over all quadrature points. However an Abaqus UMAT is only written for one quadrature poitnt. Therefore, we need to write code that implements a for loop in which the UMAT is called for each quadrature point. This is done in a C++ class that is then compiled into a shared library. The shared library is then imported into Python as in the previous examples. 
+
+In addition to the dependencies from the C++ example, we need to install a Fortran compiler. You can install it into your current conda-environment by running the following command:
+
+```bash
+mamba install gfortran -c conda-forge
+```
+
+
+After that you can create a new C++ project with the following commands:
+
+```bash
+mkdir umat
+mkdir umat/src
+touch umat/CMakeLists.txt
+touch src/main.cpp
+touch src/umat_linear_elastic.f
+```
+
+This will create a new folder called `umat` with the following structure:
+
+```bash
+umat
+├── CMakeLists.txt
+└── src
+    ├── umat_linear_elastic.f
+    └── main.cpp
+```
+
+In order to write a constitutive model which is then linked to Python, you need to define the dependencies in the `CMakeLists.txt` file. For linear algebra on small matrices, the Eigen library is used. For interfacing with Python, the pybind11 library is used:
+
+```cmake
+--8<-- "examples/umat/CMakeLists.txt"
+```
+
+
+## Writing the model
+
+TODO: Line by line explanation of the code
+
+The full source code of the model is shown below:
```

Comment:
Maybe change that to "An example for a full source code of an elasticity model in C++ is shown below:"
(also for the others)